examples/06_loading_screen.rs

```diff
@@ -0,0 +1,78 @@
+// Example 6: draw a progress bar while loading the application
+//
+// This (pretends to) load some data when starting and in between each step (eg. each texture or
+// model or whatever), it updates a progress bar.
+//
+// Alternatively, one could also do loading in a background thread or threads and do only the
+// redrawing in the main one, to separate it.
+use std::thread::sleep;
+use std::time::Duration;
+
+use quicksilver::{
+    geom::{Rectangle, Vector},
+    graphics::{Color, Graphics},
+    lifecycle::{run, EventStream, Settings, Window},
+    Result,
+};
+
+fn main() {
+    run(
+        Settings {
+            size: Vector::new(800.0, 600.0).into(),
+            title: "Square Example",
+            ..Settings::default()
+        },
+        app,
+    )
+}
+
+const STEPS: usize = 10;
+
+fn load_something() {
+    /*
+     * Here would be some actual loading of something, like textures.
+     * In this example we simply call sleep as a placeholder.
+     *
+     * TODO: Something better than sleep here would be nice.
+     * Also, there's a problem with sleeps:
+     * https://github.com/ryanisaacg/quicksilver/issues/580
+     */
+    sleep(Duration::from_secs(1));
+}
+
+fn draw_loader(window: &Window, gfx: &mut Graphics, progress: usize, total: usize) -> Result<()> {
+    gfx.clear(Color::BLACK);
+    gfx.fill_rect(
+        &Rectangle::new(Vector::new(50.0, 500.0), Vector::new(700.0, 25.0)),
+        Color::YELLOW,
+    );
+
+    let width = 700.0 * progress as f32 / total as f32;
+    gfx.fill_rect(
+        &Rectangle::new(Vector::new(50.0, 500.0), Vector::new(width, 25.0)),
+        Color::BLUE,
+    );
+
+    // Might as well draw some image above the bar too...
```

Comment:
Do you plan to come back to this or should we remove it and merge?

Reply:
I'm not sure I understand. Remove what, exactly? The comment is more like an suggestion for the reader than TODO for myself, that the progress bar in itself doesn't look very appealing, but adding an image there might help. Or is it confusing? If so, it's probably fine removing, or I can try rewording it.

Reply:
Oh, I see. We might want to add 'In a real application' before 'Might' so the reader knows they're being addressed.

Reply:
OK, I've tried to reword it a bit so its clearer.